### PR TITLE
Add JasperLake support

### PR DESCRIPTION
**Pull request by xiulipan**
Add JasperLake platform PCI ID and Kconfig and the ACPI mach table.

### CODE REVIEW

File: sound/soc/intel/common/Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@ snd-soc-acpi-intel-match-objs := soc-acpi-intel-byt-match.o soc-acpi-intel-cht-m
 	soc-acpi-intel-bxt-match.o soc-acpi-intel-glk-match.o \
 	soc-acpi-intel-cnl-match.o soc-acpi-intel-icl-match.o \
 	soc-acpi-intel-tgl-match.o soc-acpi-intel-ehl-match.o \
-	soc-acpi-intel-hda-match.o
+	soc-acpi-intel-jsl-match.o soc-acpi-intel-hda-match.o
```
Comment (plbossart) [inline sound/soc/intel/common/Makefile]: nit-pick: better to leave hda-match on a separate line.

**Comment by xiulipan** [comment]
@plbossart Updated.

File: sound/soc/sof/intel/Kconfig [hunk 2]
```diff
@@ -29,6 +29,7 @@ config SND_SOC_SOF_INTEL_PCI
 	select SND_SOC_SOF_COMETLAKE_H if SND_SOC_SOF_COMETLAKE_H_SUPPORT
 	select SND_SOC_SOF_TIGERLAKE   if SND_SOC_SOF_TIGERLAKE_SUPPORT
 	select SND_SOC_SOF_ELKHARTLAKE if SND_SOC_SOF_ELKHARTLAKE_SUPPORT
+	select SND_SOC_SOF_JASPERLAKE  if SND_SOC_SOF_JASPERLAKE_SUPPORT
```
Comment (plbossart) [inline sound/soc/sof/intel/Kconfig:32]: this looks like a bad merge, you are mixing Elkhartlake and Jasperlake?

File: sound/soc/sof/intel/Kconfig [hunk 3]
```diff
@@ -244,6 +245,21 @@ config SND_SOC_SOF_ELKHARTLAKE
           This option is not user-selectable but automagically handled by
 	  'select' statements at a higher level
 
+config SND_SOC_SOF_JASPERLAKE_SUPPORT
```
Comment (plbossart) [inline sound/soc/sof/intel/Kconfig:248]: please fix commit message

ASoC: SOF: Intel: initial support to JasperLake.

Add Kconfig, PCI ID and chip info for JSL platform.
The DSP only has 2 cores for this platform.

**Comment by xiulipan** [comment]
@plbossart commits updated.

**Pull request merged**